lib/internal/test_runner/coverage.js

```diff
@@ -359,13 +351,149 @@ function getCoverageFromDirectory(coverageDirectory) {
 
       const coverageFile = join(coverageDirectory, entry.name);
       const coverage = JSONParse(readFileSync(coverageFile, 'utf8'));
-      return coverage.result;
+
+      mergeCoverage(result, coverage.result);
     }
+
+    return ArrayFrom(result.values());
   } finally {
     if (dir) {
       dir.closeSync();
     }
   }
 }
 
+function mergeCoverage(merged, coverage) {
+  for (let i = 0; i < coverage.length; ++i) {
+    const newScript = coverage[i];
+    const { url } = newScript;
+
+    // Filter out core modules and the node_modules/ directory from results.
+    if (StringPrototypeStartsWith(url, 'node:') ||
+        StringPrototypeIncludes(url, '/node_modules/') ||
```

Comment:
should this exclusion be documented (although it is not new)

Reply:
It is documented: https://github.com/nodejs/node/blob/3e7459055bbcfe35e203896cb9afffa10db2f256/doc/api/test.md?plain=1#L390-L391